diesel_cli/src/infer_schema_internals/inference.rs

```diff
@@ -14,6 +14,12 @@ static RESERVED_NAMES: &[&str] = &[
     "type", "typeof", "unsafe", "unsized", "use", "virtual", "where", "while", "yield",
 ];
 
+fn is_reserved(name: &str) -> bool {
+    RESERVED_NAMES.contains(&name) || (
+        name.starts_with("diesel__") && !name.ends_with("_")
```

Comment:
I'm a little confused by the `ends_with` check. It seems unrelated?

Reply:
It's not strictly necessary, it's just that we append an underscore as a way to make something "not reserved", so it seemed simpler and more consistent to just treat everything ending in an underscore as not a reserved word.